pyqtorch/core/operation.py

```diff
@@ -40,20 +40,21 @@ def get_parametrized_matrix_for_operation(operation_type: str, theta: torch.Tens
 
 
 def create_controlled_matrix_from_operation(
-    operation_matrix: torch.Tensor,
+    operation_matrix: torch.Tensor, n_control_qubits: int = 1
 ) -> torch.Tensor:
     """Method which takes a 2x2 torch.Tensor and transforms it into a Controlled Operation Gate
 
     Args:
 
         operation_matrix (torch.Tensor): the type of operation which should be performed (RX,RY,RZ)
+        n_control_qubits (int): The number of control qubits used
 
     Returns:
 
         torch.Tensor: the resulting controlled gate populated by operation_matrix
     """
     mat_size = len(operation_matrix)
-    controlled_mat: torch.Tensor = torch.eye(2 * mat_size, dtype=torch.cdouble)
+    controlled_mat: torch.Tensor = torch.eye(2**n_control_qubits * mat_size, dtype=torch.cdouble)
     controlled_mat[-mat_size:, -mat_size:] = operation_matrix
     return controlled_mat
```

Comment:
The docstring says it takes a `2x2`, but isn't it actually working for any size of the operation? That's what `mat_size` does right?